pubtools/exodus/hooks/pulp.py

```diff
@@ -0,0 +1,165 @@
+import logging
+import os
+import sys
+
+import backoff
+import requests
+from pubtools.pluggy import hookimpl, pm  # pylint: disable=wrong-import-order
+from six.moves.urllib.parse import urljoin
+
+LOG = logging.getLogger("pubtools-exodus")
+
+
+class ExodusPulpHanlder:
+    def __init__(self):
+        self._gw_url = os.getenv("EXODUS_GW_URL")
+        self._gw_env = os.getenv("EXODUS_GW_ENV")
+        self._gw_crt = os.getenv("EXODUS_GW_CERT")
+        self._gw_key = os.getenv("EXODUS_GW_KEY")
+
+        self._publish = None
+        self._check_cert()
+
+    def _new_session(self):
+        out = requests.Session()
+        out.cert = (self._gw_crt, self._gw_key)
+        return out
+
+    def _do_request(self, **kwargs):
+        session = self._new_session()
+        resp = session.request(**kwargs)
+        self._unpack_response(resp)
+        return resp
+
+    def _check_cert(self):
+        """Issue request to exodus-gw to identify permissions."""
+
+        if not self._gw_crt:
+            LOG.debug("EXODUS_GW_CERT not found, authentication may fail")
+
+        if not self._gw_key:
+            LOG.debug("EXODUS_GW_KEY not found, authentication may fail")
+
+        auth_url = urljoin(self._gw_url, "/whoami")
+        resp = self._do_request(method="GET", url=auth_url)
+        context = resp.json()
+
+        for user_type, ident in (
+            ("client", "serviceAccountId"),
+            ("user", "internalUsername"),
+        ):
+            typed_ctx = context[user_type]
+            if typed_ctx["authenticated"]:
+                LOG.debug(
+                    "You are authenticated as %s %s with roles: %s",
```

Comment:
These log messages should probably be reworded now.

The original "You are authenticated as ..." messages were written for scripts that a human executes in a terminal, so the message is oriented towards them.  However we're now working on code which should be run in a Pub task and "you" is actually Pub.  Also this doesn't mention exodus at all, so it'd be confusing when mixed with other messages.

You'd probably want to rewrite this to be more neutral and also include the exodus-gw URL, like "Authenticated to exodus-gw at $URL as $NAME (roles: $ROLES)".  That would make more sense when mixed in with other log messages in a Pub task.